ts/Series/Treegraph/TreegraphSeries.ts

```diff
@@ -0,0 +1,948 @@
+import type SVGAttributes from '../../Core/Renderer/SVG/SVGAttributes';
+import type Point from '../../Core/Series/Point';
+import type PointerEvent from '../../Core/PointerEvent';
+
+import Series from '../../Core/Series/Series.js';
+import SeriesRegistry from '../../Core/Series/SeriesRegistry.js';
+import SVGRenderer from '../../Core/Renderer/SVG/SVGRenderer.js';
+import TreegraphNode from './TreegraphNode.js';
+import TreegraphPoint from './TreegraphPoint.js';
+import TU from '../TreeUtilities.js';
+import PathUtilities from '../PathUtilities.js';
+const { prototype: { symbols } } = SVGRenderer;
+const { getLevelOptions } = TU;
+const {
+    seriesTypes: {
+        treemap: TreemapSeries,
+        column: ColumnSeries
+    }
+} = SeriesRegistry;
+
+import U from '../../Core/Utilities.js';
+const { merge, pick, relativeLength, isArray } = U;
+
+import TreegraphLink from './TreegraphLink.js';
+import { Palette } from '../../Core/Color/Palettes.js';
+import { DataLabelTextPathOptions } from '../../Core/Series/DataLabelOptions.js';
+import TreegraphLayout from './TreegraphLayout.js';
+import { TreegraphSeriesLevelOptions } from './TreegraphSeriesOptions.js';
+
+interface LayoutModifiers {
+    ax: number;
+    bx: number;
+    ay: number;
+    by: number;
+}
+/* *
+ *
+ *  Class
+ *
+ * */
+
+/**
+ * The Treegraph series type.
+ *
+ * @private
+ * @class
+ * @name Highcharts.seriesTypes.treegraph
+ *
+ * @augments Highcharts.Series
+ */
+class TreegraphSeries extends TreemapSeries {
+    /* *
+     *
+     *  Static Properties
+     *
+     * */
+
+    /**
+     * A treegraph series is a diagram, which shows a relation between ancestors
+     * and descendants with a clear parent - child relation.
+     * The best examples of the dataStructures, which best reflect this chart
+     * are e.g. genealogy tree or directory structure.
+     *
+     * TODO change back the demo path
+     * @sample highcharts/demo/treegraph-chart
+     *          Treegraph Chart
+     *
+     * @extends      plotOptions.treemap
+     * @excluding    layoutAlgorithm, dashStyle, linecap, lineWidth,
+     *               negativeColor, threshold, zones, zoneAxis, colorAxis,
+     *               colorKey, compare, dataGrouping, endAgle, gapSize, gapUnit,
+     *               ignoreHiddenPoint, innerSize, joinBy, legendType, linecap,
+     *               minSize, navigatorOptions, pointRange, allowTraversingTree,
+     *               alternateStartingDirection, borderRadius, breadcrumbs,
+     *               interactByLeaf, layoutStartingDirection, levelIsConstant,
+     *               lineWidth, negativeColor, nodes, sortIndex, zoneAxis,
+     *               zones
+     *
+     * @since        next
+     * @product      highcharts
+     * @requires     modules/treemap.js
+     * @requires     modules/treegraph.js
+     * @optionparent plotOptions.treegraph
+     */
+    public static defaultOptions: TreegraphSeriesOptions = merge(
```

Comment:
Because of the size of the default options this should go into a separate file named `TreegraphSeriesDefaults.ts`.